src/lib/lessons.ts

```diff
@@ -89,12 +91,22 @@ async function loadLessonMetadataFromSanity(slug: string) {
   }
 }
 
-const deriveDataFromBaseValues = (result: any) => {
-  const http_url = `${process.env.NEXT_PUBLIC_DEPLOY_URL}${result.path}`
-  const lesson_view_url = `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1${result.path}/views`
-  const download_url = `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1${result.path}/signed_download`
+// TODO: Move this into `src/utils/lesson-metadata.ts` and add tests.
+const deriveDataFromBaseValues = ({path}: {path: string}) => {
+  if (!isEmpty(path)) {
+    invariant(
+      path.startsWith('/'),
+      'Path value must begin with a forward slash (`/`).',
+    )
```

Comment:
invariant check that the path begins with a forward slash, otherwise we'll end up with malformed URLs

Going to add this as a validation check on the Sanity schema `path` attribute as well.